### PR TITLE
feat: add commandhandling projections for address

### PR DESCRIPTION
**Pull request by pgallik**

### CODE REVIEW

File: src/ParcelRegistry.Consumer.Address/ConsumerAddressModule.cs [hunk 1]
```diff
@@ -32,6 +32,8 @@ public class ConsumerAddressModule : Module
                 RunInMemoryDb(services, loggerFactory, logger);
             }
 
+
```
Comment (emalfroy) [inline src/ParcelRegistry.Consumer.Address/ConsumerAddressModule.cs:35]: Remove whitespace

**Review by emalfroy — APPROVED** [review]

File: src/ParcelRegistry.Consumer.Address/ConsumerAddressModule.cs [hunk 2]
```diff
@@ -31,7 +31,7 @@ public class ConsumerAddressModule : Module
             {
                 RunInMemoryDb(services, loggerFactory, logger);
             }
-
+
```
Comment (ArneD) [inline src/ParcelRegistry.Consumer.Address/ConsumerAddressModule.cs:34]: whitespace

File: src/ParcelRegistry.Api.BackOffice/Infrastructure/Modules/ApiModule.cs [hunk 3]
```diff
@@ -50,23 +49,15 @@ protected override void Load(ContainerBuilder builder)
...
-                .RegisterModule(new BackOfficeModule(_configuration, _services, _loggerFactory))
                 .RegisterModule(new EditModule(_configuration))
+                .RegisterModule(new BackOfficeModule(_configuration, _services, _loggerFactory))
                 .RegisterModule(new MediatRModule())
                 .RegisterModule(new SqsHandlersModule(_configuration[SqsQueueUrlConfigKey]))
                 .RegisterModule(new TicketingModule(_configuration, _services))
                 .RegisterModule(new ConsumerAddressModule(_configuration, _services, _loggerFactory))
                 .RegisterSnapshotModule(_configuration);
-
+
```
Comment (ArneD) [inline src/ParcelRegistry.Api.BackOffice/Infrastructure/Modules/ApiModule.cs:60]: whitespace

File: src/ParcelRegistry.Consumer.Address/Infrastructure/Program.cs [hunk 4]
```diff
@@ -73,48 +77,71 @@ public static async Task Main(string[] args)
...
                                     return new Offset(offset);
                                 }
 
-                                logger.LogInformation($"Continuing {topic} from last offset.");
+                                logger.LogInformation($"BackOfficeConsumer continuing {topic} from last offset.");
                                 return null;
                             }
 
-                            var loggerFactory = container.GetRequiredService<ILoggerFactory>();
+                            // TODO: dont think we provide offset option for commandHandlingConsumer
```
Comment (ArneD) [inline src/ParcelRegistry.Consumer.Address/Infrastructure/Program.cs:88]: Let's remove this then :)

File: src/ParcelRegistry.Consumer.Address/Migrations/20221121094051_Initial.Designer.cs [hunk 5]
```diff
@@ -48,7 +48,7 @@ protected override void BuildTargetModel(ModelBuilder modelBuilder)
                     b.ToTable("ProjectionStates", "ParcelRegistryConsumerAddress");
                 });
 
-            modelBuilder.Entity("ParcelRegistry.Consumer.Address.AddressConsumerItem", b =>
+            modelBuilder.Entity("ParcelRegistry.BackOfficeConsumer.Address.AddressConsumerItem", b =>
```
Comment (ArneD) [inline src/ParcelRegistry.Consumer.Address/Migrations/20221121094051_Initial.Designer.cs:51]: you can't do this, migrations are already executed.

File: src/ParcelRegistry.Consumer.Address/Projections/CommandHandlingKafkaProjection.cs [hunk 6]
```diff
@@ -0,0 +1,158 @@
...
+        }
+
+        private static Provenance FromProvenance(Contracts.Provenance provenance) =>
+            new Provenance(
+                InstantPattern.General.Parse(provenance.Timestamp).GetValueOrThrow(),
+                Enum.Parse<Application>(provenance.Application),
+                new Reason(provenance.Reason),
+                new Operator(string.Empty),
+                Enum.Parse<Modification>(provenance.Modification),
+                Enum.Parse<Organisation>(provenance.Organisation));
```
Comment (ArneD) [inline src/ParcelRegistry.Consumer.Address/Projections/CommandHandlingKafkaProjection.cs:156]: Always `DigitaalVlaanderen`

File: src/ParcelRegistry.Consumer.Address/appsettings.json [hunk 7]
```diff
@@ -14,9 +14,10 @@
     "BootstrapServers": "localhost:29092/"
   },
 
-  "AddressConsumerGroupSuffix": "",
+  "BackOfficeConsumerGroupSuffix": "",
```
[the remark this answers is not in the document]
Comment (ArneD) [reply src/ParcelRegistry.Consumer.Address/appsettings.json:17]: Yes!

File: src/ParcelRegistry/Parcel/Events/ParcelAddressWasDetachedBecauseAddressWasRejected.cs [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+    using System.Collections.Generic;
+    using System.Linq;
+    using Be.Vlaanderen.Basisregisters.EventHandling;
+    using Be.Vlaanderen.Basisregisters.GrAr.Common;
+    using Be.Vlaanderen.Basisregisters.GrAr.Provenance;
+    using Newtonsoft.Json;
+
+    [EventTags(Tag.Address)]
+    [EventName(EventName)]
+    [EventDescription("Er werd een adres ontkoppeld van het perceel.")]
```
Comment (ArneD) [inline src/ParcelRegistry/Parcel/Events/ParcelAddressWasDetachedBecauseAddressWasRejected.cs:13]: Text should change to reflect the name (all new events)

File: src/ParcelRegistry.Consumer.Address/Infrastructure/Program.cs [hunk 9]
```diff
@@ -73,48 +78,63 @@ public static async Task Main(string[] args)
...
 
-                                CancellationTokenSource.Cancel();
+                            await Task.WhenAny(consumerTasks);
 
-                                Log.Error($"Consumer task stopped with status: {consumerTask.Status}");
-                            }
+                            CancellationTokenSource.Cancel();
 
-                            Log.Error("The consumer was terminated");
+                            Log.Error("The Address Consumers were terminated");
```
Comment (ArneD) [inline src/ParcelRegistry.Consumer.Address/Infrastructure/Program.cs:137]: Can this be looped? It's interesting to know which one crashed (faulted state) and which ones were cancelled.

File: src/ParcelRegistry.Consumer.Address/Projections/CommandHandlingKafkaProjection.cs [hunk 10]
```diff
@@ -0,0 +1,178 @@
...
+
+                    foreach (var relation in relations)
+                    {
+                        var command = new DetachAddressBecauseAddressWasRemoved(
+                            new ParcelId(relation.ParcelId),
+                            new AddressPersistentLocalId(message.AddressPersistentLocalId),
+                            FromProvenance(message.Provenance));
+                        await commandHandler.Handle(command, ct);
+                    }
+                }
```
Comment (ArneD) [inline src/ParcelRegistry.Consumer.Address/Projections/CommandHandlingKafkaProjection.cs:43]: What if `message.Status == 'Retired'`?

**Review by ArneD — CHANGES_REQUESTED** [review]

**Review by ArneD — COMMENTED** [review]
Also don't forget to seal classes by default.

File: src/ParcelRegistry.Consumer.Address/Infrastructure/Program.cs (hunk 9 above)
Comment (pgallik) [reply src/ParcelRegistry.Consumer.Address/Infrastructure/Program.cs:137]: Still need to do this

**Review by ArneD — APPROVED** [review]

**Pull request merged**